### DISCUSSION TITLE
Discordjs Slash Command - Option with Dynamic time

### DISCUSSION THREAD

**Discussion by moosah-edge** [discordjs/discord.js#6881] [Q&A] [answered]
Hi all,

I have a slash command that asks users to select a date and time. 

I was hoping to make the date and time dynamic, so that the first Date option would always be the current day, and the first Time option would be the current Hour as these are usually the most likely options.  This works great when the bot starts, however I quickly learned the values seem to be assigned when the command is deployed, not when a user initiates the command, so the date and time values are not updating as time progresses. 

It is possible to restart the bot which forces the values to update, however is there any method available within discordjs to have command values evaluated at runtime?  

Snippet from the command below, which essentially displays options like:
Date:  Today, Yesterday, 23/10/2021, 22/10/2021, 21/10/2021...
Time:  9am, 8am, 7am, 6am...

Thanks all.

```
commands = [
	new SlashCommandBuilder()
		.setName('task')
		.setDescription('Submit a completed task')
		.addUserOption(option =>
			option.setName('member')
				.setDescription('Your name')
				.setRequired(true))
		.addStringOption(option =>
			option.setName('date-completed')
				.setDescription('Date completed')
				.setRequired(true)
				.addChoice( 'Today', (new Date().getFullYear()+'/'+(new Date().getMonth()+1)+'/'+(new Date().getDate())) )
				.addChoice( 'Yesterday', (new Date().getFullYear()+'/'+(new Date().getMonth()+1)+'/'+(new Date().getDate()-1)) )
				.addChoice( (new Date().getDate()-2)+'/'+(new Date().getMonth()+1)+'/'+(new Date().getFullYear()), (new Date().getFullYear()+'/'+(new Date().getMonth()+1)+'/'+(new Date().getDate()-2)) )
				.addChoice( (new Date().getDate()-3)+'/'+(new Date().getMonth()+1)+'/'+(new Date().getFullYear()), (new Date().getFullYear()+'/'+(new Date().getMonth()+1)+'/'+(new Date().getDate()-3)) ))
		.addStringOption(option =>
			option.setName('time-completed')
				.setDescription('Time completed. Select the closest hour')
				.setRequired(true)
				.addChoice( formatAMPM(new Date().addHours(1)) , format24Hours(new Date().addHours(1)) )
				.addChoice( formatAMPM(new Date().addHours(0)) , format24Hours(new Date().addHours(0)) )
				.addChoice( formatAMPM(new Date().addHours(-1)) , format24Hours(new Date().addHours(-1)) )
				.addChoice( formatAMPM(new Date().addHours(-2)) , format24Hours(new Date().addHours(-2)) )
...
```

**Comment by moosah-edge** [comment]
FWIW, as a workaround I've created a cron job to update the commands every hour

**Comment by almostSouji** [comment, marked as answer]
Dynamically populating choices at the time the user uses the command will be possible with the autocomplete feature https://github.com/discordjs/discord.js/pull/6672. This will be available in the library with the next minor version 13.3 after Discord confirmed the API to be stable by merging the relevant API doc pull request.
